src/winconpty/winconpty.cpp

```diff
@@ -133,21 +133,20 @@ HRESULT _CreatePseudoConsole(const HANDLE hToken,
     RETURN_IF_WIN32_BOOL_FALSE(SetHandleInformation(signalPipeConhostSide.get(), HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT));
 
     // GH4061: Ensure that the path to executable in the format is escaped so C:\Program.exe cannot collide with C:\Program Files
-    auto pwszFormat = L"\"%s\" --headless %s%s%s--width %hu --height %hu --signal 0x%x --server 0x%x";
     // This is plenty of space to hold the formatted string
     wchar_t cmd[MAX_PATH]{};
     const BOOL bInheritCursor = (dwFlags & PSEUDOCONSOLE_INHERIT_CURSOR) == PSEUDOCONSOLE_INHERIT_CURSOR;
     const BOOL bResizeQuirk = (dwFlags & PSEUDOCONSOLE_RESIZE_QUIRK) == PSEUDOCONSOLE_RESIZE_QUIRK;
     swprintf_s(cmd,
                MAX_PATH,
-               pwszFormat,
+               L"\"%s\" --headless %s%s--width %hd --height %hd --signal 0x%tx --server 0x%tx",

```

Comment:
i see you changed additional format parameters here. can you quickly dip into what `tx` is versus `x`? we've had it the one way for like five years ;P

Reply:
`x` is an `unsigned int` sized parameter and so it's wrong to pass a `void*` sized type. The only reason it works is because on x86 the stack word size is 8 bytes and so the alignment of those vararg parameters matches the pointer size. I'm sure it's the same on ARM. Additionally, handle values aren't usually very large.

C99 introduced 3 new prefixes:
* `z`: `size_t`
* `j`: `intmax_t`
* `t`: `ptrdiff_t`

In this case passing `t` is the right choice. `z` would also be okay, but only because current architectures have `ptrdiff == size`.